bundles/framework/myplaces3/ButtonHandler.js

```diff
@@ -0,0 +1,417 @@
+                        drawMode: 'point',
+                        shape: 'Point',
+                        drawOptions: drawOptions
+                    });
+                }
+            },
+            'line': {
+                iconCls: 'myplaces-draw-line',
+                tooltip: '',
+                sticky: true,
+                callback: function () {
+                    me.sendDrawRequest({
+                        drawMode: 'line',
+                        shape: 'LineString',
+                        drawOptions: drawOptions
+                    });
+                }
+            },
+            'area': {
+                iconCls: 'myplaces-draw-area',
+                tooltip: '',
+                sticky: true,
+                callback: function () {
+                    me.sendDrawRequest({
+                        drawMode: 'area',
+                        shape: 'Polygon',
+                        drawOptions: drawOptions
+                    });
+                }
+            }
+        };
+        this.templateGuide = jQuery('<div><div class="guide"></div>' +
+            '<div class="buttons">' +
+            '<div class="cancel button"></div>' +
+            '<div class="finish button"></div>' +
+            '</div>' +
+            '</div>');
+
+        this.templateHelper = jQuery(
+            '<div class="drawHelper">' +
+            '<div class="infoText"></div>' +
+            '<div class="measurementResult"></div>' +
+            '</div>'
+        );
+    }, {
+        __name: 'MyPlacesButtonHandler',
+        /**
+         * @method getName
+         * @return {String} the name for the component
+         */
+        getName: function () {
+            return this.__name;
+        },
+        /**
+         * @method init
+         * implements Module protocol init method
+         */
+        init: function () {
+            var user = Oskari.user();
+            // different tooltip for guests - "Please log in to use"
+            var guestPostfix = ' - ' + this.loc('guest.loginShort'),
+                tool,
+                tooltip;
+            for (tool in this.buttons) {
+                if (this.buttons.hasOwnProperty(tool)) {
+                    tooltip = this.loc('tools.' + tool + '.tooltip');
+                    if (!user.isLoggedIn()) {
+                        tooltip = tooltip + guestPostfix;
+                    }
+                    this.buttons[tool].tooltip = tooltip;
+                }
+            }
+        },
+        /**
+         * @method start
+         * implements Module protocol start methdod
+         */
+        start: function () {
+            var me = this,
+                sandbox = me.instance.sandbox,
+                p,
+                tool,
+                measureTool;
+            sandbox.register(me);
+
+            // request toolbar to add buttons
+            var reqBuilder = sandbox.getRequestBuilder('Toolbar.AddToolButtonRequest');
```

Comment:
Use Oskari.requestBuilder() instead of sandbox.getRequestBuilder(). Remove the commented out code.